packages/settingregistry/src/plugin-schema.json

```diff
@@ -15,26 +33,103 @@
       "type": "string",
       "default": ""
     },
-    "jupyter.lab.setting-icon-label": { "type": "string", "default": "Plugin" },
+    "jupyter.lab.setting-icon-label": {
+      "type": "string",
+      "default": "Plugin"
+    },
     "jupyter.lab.shortcuts": {
-      "items": { "$ref": "#/definitions/shortcut" },
+      "items": {
+        "$ref": "#/definitions/shortcut"
+      },
       "type": "array",
       "default": []
     },
-    "jupyter.lab.transform": { "type": "boolean", "default": false }
+    "jupyter.lab.transform": {
+      "type": "boolean",
+      "default": false
+    }
   },
   "definitions": {
+    "menu": {
+      "properties": {
+        "disabled": {
+          "type": "boolean",
+          "default": false
+        },
+        "id": {
+          "type": "string"
+        },
+        "items": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/menuItem"
+          }
+        },
+        "label": {
+          "type": "string"
+        },
+        "rank": {
+          "type": "number"
+        }
+      },
+      "required": ["id"],
+      "type": "object"
+    },
+    "menuItem": {
+      "properties": {
+        "args": {
+          "type": "object"
+        },
+        "command": {
+          "type": "string"
+        },
+        "disabled": {
+          "type": "boolean",
+          "default": false
+        },
+        "type": {
+          "type": "string",
+          "enum": ["command", "submenu", "separator"],
+          "default": "command"
+        },
+        "rank": {
+          "type": "number"
```

Comment:
again consider `minimum`

Reply:
I had to set it to minimum 0. Otherwise it will change existing rank.